File: A/1092/1092.py
n = int(input())
letters = "abcdefghijklmnopqrstuvwxyz"

for _ in range(n):
    l, t = map(int, input().split())
    nova = letters[:t]   # equivalente a substr(0, t)
    
    final = ""
    rep = 0
    
    for _ in range(l):
        if rep == t:
            rep = 0
        final += nova[rep]
        rep += 1
    
    print(final)

File: A/1092/1092.cpp
#include <iostream>

int main() {
	int n, l, t;
	std::cin >> n;
	std::string letters = "abcdefghijklmnopqrstuvwxyz";
	while (n--) {
		std::cin >> l >> t;
		int rep = 0;
		std::string final;
		std::string nova = letters.substr(0,t);
		while(l--) {
			if(rep==t) rep = 0;
			final += nova[rep];
			rep++;
		}
		std::cout << final << std::endl;
	}
	return 0;
}
